Ghidra/Features/Base/ghidra_scripts/FixPascalCallingConvention.java

```diff
@@ -0,0 +1,281 @@
+// on function parameters and changes the stack reference for left-to-
+// right stacking.  On the way, it also ensures that all Thunks are
+// also converted.  This applies to Windows 16-bit apps.
+//
+//@category Repair
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import ghidra.app.script.GhidraScript;
+import ghidra.program.model.address.Address;
+import ghidra.program.model.address.AddressSet;
+import ghidra.program.model.data.GenericCallingConvention;
+import ghidra.program.model.listing.Function;
+import ghidra.program.model.listing.FunctionIterator;
+import ghidra.program.model.listing.FunctionManager;
+import ghidra.program.model.listing.Parameter;
+import ghidra.program.model.listing.ParameterImpl;
+import ghidra.program.model.listing.VariableStorage;
+import ghidra.program.model.pcode.Varnode;
+import ghidra.program.model.symbol.SourceType;
+import ghidra.util.exception.DuplicateNameException;
+import ghidra.util.exception.InvalidInputException;
+
+public class FixPascalCallingConvention extends GhidraScript {
+
+	private static final String PASCAL16FAR = "__pascal16far";
+
+	private int cntFnsUpdated;
+	private int cntParamsTotal;
+	private int cntParamsReplaced;
+	private int cntConvertionTotal;
+	private int cntConvertionChanged;
+
+	private List<String> warningMessages = new ArrayList<String>();
+
+	@Override
+	public void run() throws Exception {
+
+		// reset for this run
+		cntConvertionTotal = 0;
+		cntConvertionChanged = 0;
+		cntFnsUpdated = 0;
+		cntParamsTotal = 0;
+		cntParamsReplaced = 0;
+
+		warningMessages.clear();
+
+		if (currentProgram != null) {
+			FunctionManager fnMgr = currentProgram.getFunctionManager();
+			if (fnMgr == null) {
+				return;
+			}
+
+			Iterator<Function> functions = null;
+
+			// add function if pointing at its entry point ...
+			if (isAddressAtFunctionStart(fnMgr, currentAddress)) {
+				functions = fnMgr.getFunctionsOverlapping(new AddressSet(currentAddress));
+			}
+
+			// ... if not pointing at start of any function, try those currently selected
+			// ...
+			if (((functions == null) || !functions.hasNext()) && (currentSelection != null)) {
+				functions = fnMgr.getFunctionsOverlapping(currentSelection);
+			}
+
+			// ... otherwise, choose them all
+			if ((functions == null) || !functions.hasNext()) {
+				functions = fnMgr.getFunctions(true);
+			}
+
+			// update details
+			doRun(functions);
+
+			// build popup information
+			StringBuffer buf = new StringBuffer("Update " + cntConvertionChanged
+					+ " calling convertions out of " + cntConvertionTotal + " to " + PASCAL16FAR
+					+ ".\n\nUpdated " + cntFnsUpdated + " functions having " + cntParamsReplaced
+					+ " parameters reversed out of " + cntParamsTotal + " parameters found.");
+			Iterator<String> messages = warningMessages.iterator();
+			while (messages.hasNext()) {
+				buf.append("\n\n").append(messages.next());
+			}
+
+			popup(buf.toString());
```

Comment:
`StringBuffer` is for when synchronization is required, otherwise you should use `StringBuilder`. It actually isn't necessary to use either here. After the concatination of the original message you c
```suggestion
			String buf = "Update " + cntConvertionChanged
					+ " calling convertions out of " + cntConvertionTotal + " to " + PASCAL16FAR
					+ ".\n\nUpdated " + cntFnsUpdated + " functions having " + cntParamsReplaced
					+ " parameters reversed out of " + cntParamsTotal + " parameters found.";
			popup(buf + String.join("\n\n", warningMessages);
```